x-pack/solutions/security/plugins/security_solution/public/detection_engine/rule_creation_ui/components/rule_preview/preview_histogram.tsx

```diff
@@ -97,11 +98,11 @@ const PreviewHistogramComponent = ({
   const previousPreviewId = usePrevious(previewId);
   const previewQueryId = `${ID}-${previewId}`;
   const previewEmbeddableId = `${previewQueryId}-embeddable`;
-  const { responses: visualizationResponses } = useVisualizationResponse({
+  const { tables } = useVisualizationResponse({
     visualizationId: previewEmbeddableId,
   });
```

Comment:
Since this seems to be a core use case for the hook (as you stated in the PR description), did you consider adding `totalCount` as part of the hook's response, instead of making it external? Then the hook would not have to expose the table abstraction completely for these "total" consumers:

```suggestion
  const { getTotalHits } = useVisualizationResponse({
    visualizationId: previewEmbeddableId,
  });
  
  const totalCount = getTotalHits();
```

Reply:
hey @rylnd that's a great point.

I did indeed consider it initially. But I am actually assuming here that folks are ok with summing all lens layer totalCounts. For example you can have 2 separate chart renderings with separate labels that you want to control independently. And for that you can access `tables[layerId].meta.statistics.totalCount` directly if you have a layerId at hand for xy charts which you do because it's in the lens attributes file. Or create your own extraction logic.

And since I wasn't sure what is the better way and didn't want to decide the use case for all teams I decided to extract that logic out right next and let others choose to use it if it makes sense for their use case or choose to access table directly or with a different parser. 

Also with exposure of `tables` I wanted to emphasize the fact that we now work with datatable structure instead of regular req and response objects. They are very different in their representations and datables have nuances.

And on a minor note I would stay away from naming it getTotalHits() because it uses the language of non datatable response aka `response.hits.total` shape which can be confusing because the expected shapes don't match.  `response.hits.total` can be both a number and an object while totalCounts is always a number. And I know we can fix it by enforcing the types, but that's not the point. 

Domain language is the point and to me `getTotalHits` seems to be associated with regular request response shape. I also saw something similar used in your code with `getTotalHitsValue` which kind of corroborates the point.

Basically I would want to avoid confusing folks that might think this is a regular response shape's total hits so if their next move is to access more details from response they could start searching for entrypoint to aggregations and checking buckets in the response itself and will instead find tables which is breaking the expectation. To me it's always good to be as expressive and unambiguous in your api description as possible wdyt?

And finally we could just incorporate totalCounts into tables response by introducing a container of tables like 

```json
tables: {
  tables: {  [layer1]: {...}}
  meta: { statistics: { totalCounts: 999}}
}
```

To just mirror the structure of each individual layer meta totalcounts and get rid of the external helper.  To me this is a little more idiomatic

Let me know if my thoughts make sense to you.

Reply:
hey @rylnd that's a great point.

I did indeed consider it initially. But I am actually assuming here that folks are ok with summing all lens layer totalCounts. For example you can have 2 separate chart renderings with separate labels that you want to control independently. And for that you can access `tables[layerId].meta.statistics.totalCount` directly if you have a layerId at hand for xy charts which you do because it's in the lens attributes file. Or create your own extraction logic.

And since I wasn't sure what is the better way and didn't want to decide the use case for all teams I decided to extract that logic out right next and let others choose to use it if it makes sense for their use case or choose to access table directly or with a different parser. 

Also with exposure of `tables` I wanted to emphasize the fact that we now work with datatable structure instead of regular req and response objects. They are very different in their representations and datables have nuances.

And on a minor note I would stay away from naming it getTotalHits() because it uses the language of non datatable response aka `response.hits.total` shape which can be confusing because the expected shapes don't match.  `response.hits.total` can be both a number and an object while totalCounts is always a number. And I know we can fix it by enforcing the types, but that's not the point. 

Domain language is the point and to me `getTotalHits` seems to be associated with regular request response shape. I also saw something similar used in your team code with `getTotalHitsValue` which kind of corroborates the point.

Basically I would want to avoid confusing folks that might think this is a regular response shape's total hits so if their next move is to access more details from response they could start searching for entrypoint to aggregations and checking buckets in the response itself and will instead find tables which is breaking the expectation. To me it's always good to be as expressive and unambiguous in your api description as possible wdyt?

And finally if we still decide to internalize it we could just incorporate totalCounts into tables response by introducing a wrapper for tables like 

```ts
{ 
  tables: {
    tables: {  [layer1]: {...}}
    meta: { statistics: { totalCounts: 999}}
  }
}
```

To just mirror the structure of each individual layer meta totalCounts and get rid of the external helper.  To me this is a little more idiomatic. Let me know wdyt?

Reply:
Those are all fair points! I was mostly trying to minimize the changes needed here, particularly to the tests (which were closely tied to the implementation far before this PR). I don't disagree with returning `tables`, nor about exporting the helper method for more "manual" purposes.

Mainly I was just trying to make the "I only care about the total count" use case simpler and easier, since it seems to be a common one. Having to import and invoke an extra function isn't terrible, but as you've seen it makes more work (and more implementation details) in the tests, as well.

IMO the cleanliness of moving that logic into the hook is well worth the cost; as long as the total calculation is deferred, the only real "cost" is memory for the function definition: consumers are free to call or ignore it as needed. I suspect this is where we might disagree, which is fine.

P.S. I admit that `getTotalHits` was not a great example name, especially since `getTotalCountFromTables` is already established 😉.

Reply:
I absolutely appreciate your points as well.

> but as you've seen it makes more work (and more implementation details) in the tests, as well.
  
I agree with this.

> the cleanliness of moving that logic into the hook is well worth the cost

I agree as well. Not opposing at all. Was just highlighting how I came to that conclusion in those circumstances.

>  the only real "cost" is memory for the function definition

As you correctly inferred I do disagree but again because imho miniscule memory impact does not justify adding an arguably preventable interface variation. We already have totals as numbers, so having a total of totals as a number is a more natural expectation, especially given that the calculation itself is going to be wrapped in useMemo and computational impact will be reduced further. 

And lastly Imho we shouldn't be even debating this, this total totalcount should come out of the box as part of tablesAdapter. But since it's not I think what I suggested should work. I will try and if it doesn't I will use your suggestion.

Thanks

Reply:
Done